### PR TITLE
Add heatmap vis type

### LINKED ISSUE #13871: Add heatmap vis type
Depends on https://github.com/influxdata/vis/issues/25

- [ ] Add new view type to backend (will need to be deployed before any UI changes)
- [ ] Use [vis library heatmap layer](https://influxdata.github.io/vis/?path=/story/xy-plot--heatmap)
- [ ] Support selecting which `x` and `y` columns are used in the visualization
- [ ] Support selecting a sequential color schemes to use in the heatmap (will need some help from @alexpaxton or @mavarius for this)
- [ ] Support labeling x and y axis
- [ ] Support x/y axis prefix/suffixes
- [ ] Support setting x/y axis domains

### PR DESCRIPTION
**Pull request by chnn**
Closes #13871

Adds a heatmap / 2D histogram visualization type:

selected

The ability to create heatmaps is feature flagged in the UI, which will help us roll out the API updates to cloud before the UI updates.

To enable the feature, open the dev tools and type:

```
influx.toggleFeature('heatmap')
```

Then refresh the page.

The heatmaps will recompute the binning statistic on zoom and resize:

rebinning

Heatmaps support a variety of visualization options:

options

The line and histogram options have been tweaked for consistency between all visualization types, though there's still lots of work to do in cleaning up the vis options:

options_side_by_side

### CODE REVIEW

File: ui/src/shared/components/ColorSchemeDropdownItem.scss [hunk 1]
```diff
@@ -1,10 +1,10 @@
-.color-scheme-dropdown--item {
+.color-scheme-dropdown-item {
   display: flex;
   justify-content: flex-start;
   align-items: center;
 }
 
-.color-scheme-dropdown--swatches {
+.color-scheme-dropdown-item--swatches {
```
Comment (alexpaxton) [inline ui/src/shared/components/ColorSchemeDropdownItem.scss]: Naisss

File: ui/src/shared/components/ColorSchemeDropdownItem.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    background: `linear-gradient(to right, ${stops})`,
+  }
+}
+
+interface Props {
+  name: string
+  colors: string[]
+}
+
+const ColorSchemeDropdownItem: FunctionComponent<Props> = ({name, colors}) => {
```
Comment (alexpaxton) [inline ui/src/shared/components/ColorSchemeDropdownItem.tsx]: Good to see more function components

File: ui/src/shared/components/XYContainer.tsx [hunk 3]
```diff
@@ -120,7 +124,7 @@ const XYContainer: FunctionComponent<Props> = ({
   }
 
   return (
-    <div className="xy-container">
+    <div className="vis-plot-container">
```
Comment (alexpaxton) [inline ui/src/shared/components/XYContainer.tsx]: It's all in the details

File: ui/src/shared/components/cells/Cell.scss [hunk 4]
```diff
@@ -29,6 +29,10 @@ $cell--header-size: 36px;
   .time-machine-tables {
     padding-top: 0;
   }
+
+  .vis-plot-container {
+    padding-top: 5px;
```
Comment (alexpaxton) [inline ui/src/shared/components/cells/Cell.scss]: Small nit, but would be nice to see use of the global margin variables, which would shoe horn this into either `4px ` or `6px`

`$ix-marg-a`
or
`$ix-marg-a + $ix-border`

File: ui/src/shared/components/cells/Cell.scss [hunk 5]
```diff
@@ -173,6 +177,12 @@ $cell--header-size: 36px;
   }
 }
 
+.vis-plot-container {
+  width: 100%;
+  height: 100%;
+  padding: 15px;
```
Comment (alexpaxton) [inline ui/src/shared/components/cells/Cell.scss]: Same comment about variables, there's one that's `16px` I think it's `$ix-marg-c`

File: ui/src/shared/components/cells/Cell.scss (hunk 4 above)
Comment (chnn) [reply ui/src/shared/components/cells/Cell.scss]: Ah sorry, this is like the millionth time you've told me this. Will update!

**Pull request merged**